models/demos/t3000/llama2_70b/tt/llama_generation.py

```diff
@@ -305,8 +305,16 @@ def _process_logits(self, tt_logits):
 
 
 def get_padded_prefill_len(seq_len):
-    # Llama supports power of 2 lengths that are greater than 32
-    return max(32, nearest_pow_2(seq_len))
+    """
+    If seq_len is less than 32, pad to 32
+    If seq_len is more than 32, pad to whichever is smaller: a power of 2 or a multiple of 1024
+    TODO: Generalize for max_mm_seq_len different from 1024
+    """
+    pow_2_pad = nearest_pow_2(seq_len)
+    mult_1024_pad = 1024 * math.ceil(pow_2_pad / 1024)
```

Comment:
Should this be seq_len/1024 for nearest multiple of 1024?

Reply:
Thanks! Took another look, and realized that the padding function would never return 32 even if seq_len <= 32. Should be fixed, please check my work.

Reply:
Looks good, thanks